File: Core/NetworkScanner/ServiceCatalog.swift
import Foundation

enum ServiceCatalog {
    static let defaultPorts = [22, 80, 443, 445, 548, 5900, 3389, 8080]

    static func service(for port: Int) -> NetworkService {
        switch port {
        case 22:
            return NetworkService(port: port, name: "SSH", scheme: "ssh")
        case 80:
            return NetworkService(port: port, name: "HTTP", scheme: "http")
        case 443:
            return NetworkService(port: port, name: "HTTPS", scheme: "https")
        case 445:
            return NetworkService(port: port, name: "SMB", scheme: "smb")
        case 548:
            return NetworkService(port: port, name: "AFP", scheme: "afp")
        case 5900:
            return NetworkService(port: port, name: "VNC", scheme: "vnc")
        case 3389:
            return NetworkService(port: port, name: "RDP", scheme: "rdp")
        case 8080:
            return NetworkService(port: port, name: "HTTP-alt", scheme: "http")
        default:
            return NetworkService(port: port, name: "TCP", scheme: nil)
        }
    }
}

File: Windows/LanScope.Core/Services/ServiceCatalog.cs
using LanScope.Core.Models;

namespace LanScope.Core.Services;

public static class ServiceCatalog
{
    private static readonly IReadOnlyDictionary<int, NetworkService> Services =
        new Dictionary<int, NetworkService>
        {
            [22] = new(22, "SSH", "ssh"),
            [80] = new(80, "HTTP", "http"),
            [443] = new(443, "HTTPS", "https"),
            [445] = new(445, "SMB", "smb"),
            [548] = new(548, "AFP", "afp"),
            [3389] = new(3389, "RDP", "rdp"),
            [5900] = new(5900, "VNC", "vnc"),
            [8080] = new(8080, "HTTP-alt", "http")
        };

    public static NetworkService ForPort(int port) =>
        Services.TryGetValue(port, out var service) ? service : new NetworkService(port, $"TCP {port}");
}
